app/tests/TestCase/Services/AuthorizationEdgeCasesTest.php

```
@@ -0,0 +1,528 @@
+     * 
+     * Eirik (2875) has member_role 362 (Greater Officer of State) that was revoked (revoker_id = 1073)
+     * and expired on 2025-08-30. This role should not grant any permissions.
+     *
+     * @return void
+     */
+    public function testRevokedRoleDoesNotGrantPermissions(): void
+    {
+        $eirik = $this->Members->get(self::TEST_MEMBER_EIRIK_ID);
+
+        // Load permissions (should filter out revoked roles)
+        $permissions = $eirik->getPermissions();
```

PHPMD (2.15.0)

Avoid unused local variables such as '$permissions'. (undefined)

(UnusedLocalVariable)